Source/radiation/MGFLD.cpp

```diff
@@ -925,30 +927,78 @@ void Radiation::MGFLD_compute_rosseland(FArrayBox& kappa_r, const FArrayBox& sta
 
   const Box& kbox = kappa_r.box();
 
-#pragma gpu box(kbox) sync
-  ca_compute_rosseland(AMREX_INT_ANYD(kbox.loVect()), AMREX_INT_ANYD(kbox.hiVect()),
-                       BL_TO_FORTRAN_ANYD(kappa_r),
-                       BL_TO_FORTRAN_ANYD(state),
-                       0, nGroups-1, nGroups);
+  auto state_arr = state.array();
+  auto kpr = kappa_r.array();
+
+  GpuArray<Real, NGROUPS> nugroup_loc;
+  for (int g = 0; g < NGROUPS; ++g) {
+      nugroup_loc[g] = nugroup[g];
+  }
 
+  amrex::ParallelFor(kbox,
+  [=] AMREX_GPU_HOST_DEVICE (int i, int j, int k) noexcept
+  {
+      Real rho = state_arr(i,j,k,URHO);
+      Real temp = state_arr(i,j,k,UTEMP);
+      Real Ye;
+      if (NumAux > 0) {
+          Ye = state_arr(i,j,k,UFX);
+      } else {
+          Ye = 0.e0_rt;
+      }
+
+      Real kp, kr;
+      bool comp_kp = false;
+      bool comp_kr = true;
+
+      for (int g = 0; g < NGROUPS; ++g) {
+          opacity(kp, kr, rho, temp, Ye, nugroup_loc[g], comp_kp, comp_kr);
+          kpr(i,j,k,g) = kr;
+      }
+  });
+  Gpu::synchronize();
```

Comment:
why does this need a Gpu::synchronize?

Reply:
oh I see -- it was there previously

Reply:
It was there previously because this is the Fab interface, not the MultiFab interface, and when you only have the Fab at function scope you don't know if it's being used in a fashion safe for asynchrony. This is a pain point I want to clean up after I get everything converted to C++.